lib/bitcoin/BitcoinClient.ts

```diff
@@ -13,33 +17,91 @@ export default class BitcoinClient implements IBitcoinClient {
   /** Bitcoin peer's RPC basic authorization credentials */
   private readonly bitcoinAuthorization?: string;
 
+  /** Wallet private key */
+  private readonly privateKey: PrivateKey;
+  private readonly privateKeyAddress: Address;
+
   constructor (
     private bitcoinPeerUri: string,
     bitcoinRpcUsername: string | undefined,
     bitcoinRpcPassword: string | undefined,
+    bitcoinWalletImportString: string,
     private requestTimeout: number,
     private requestMaxRetries: number) {
 
+    // Bitcore has a type file error on PrivateKey
+    try {
+      this.privateKey = (PrivateKey as any).fromWIF(bitcoinWalletImportString);
+    } catch (error) {
+      throw new Error(`Failed creating private key from '${bitcoinWalletImportString}': ${error.message}`);
+    }
+
+    this.privateKeyAddress = this.privateKey.toAddress();
+
     if (bitcoinRpcUsername && bitcoinRpcPassword) {
       this.bitcoinAuthorization = Buffer.from(`${bitcoinRpcUsername}:${bitcoinRpcPassword}`).toString('base64');
     }
   }
 
-  public async broadcastTransaction (transaction: Transaction): Promise<boolean> {
+  public async initialize (): Promise<void> {
+
+    console.debug(`Checking if bitcoin contains a wallet for ${this.privateKeyAddress}`);
+    if (!await this.walletExists(this.privateKeyAddress.toString())) {
+      console.debug(`Configuring bitcoin peer to watch address ${this.privateKeyAddress}. This can take up to 10 minutes.`);
+
+      const publicKeyAsHex = this.privateKey.toPublicKey().toBuffer().toString('hex');
+      await this.importPublicKey(publicKeyAsHex, true);
+    } else {
+      console.debug('Wallet found.');
+    }
+  }
+
+  /**
+   * generates a private key in WIF format
+   * @param network Which bitcoin network to generate this key for
+   */
+  public static generatePrivateKey (network: 'mainnet' | 'livenet' | 'testnet' | undefined): string {

```

Comment:
Note that `bitcoin.ts` in ION repo uses this method, but I am actually fine if you simply delete that line there as opposed to expose this class or method in the export just for that purpose.

Reply:
Hmmm .. didn't realize that. I feel that usage is helpful. Let me think about this one.

Reply:
I'm adding a new public method in the bitcoinprocessor which calls this one.